### PR TITLE
BUSCO: additional outputs

### PR DESCRIPTION
**Pull request by priyanka-surana**
This commits adds additional outputs.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/busco/meta.yml [hunk 1]
```diff
@@ -56,6 +56,14 @@ output:
       type: directory
       description: BUSCO lineage specific output
       pattern: "*-busco"
+  - full_table:
+      type: file
+      description: Full BUSCO results table
+      pattern: "*-busco/*/run_*/full_table.tsv"
```
Comment (muffato) [inline modules/nf-core/busco/meta.yml:62]: Only the name needs to be documented
```suggestion
      pattern: "full_table.tsv"
```

File: modules/nf-core/busco/meta.yml [hunk 2]
```diff
@@ -56,6 +56,14 @@ output:
...
+  - seq_dir:
+      type: dir
+      description: BUSCO sequence directory
+      pattern: "*-busco/*/run_*/busco_sequences"
```
Comment (muffato) [inline modules/nf-core/busco/meta.yml:66]: Only the name needs to be documented
```suggestion
      pattern: "busco_sequences"
```

File: modules/nf-core/busco/main.nf [hunk 3]
```diff
@@ -14,11 +14,13 @@ process BUSCO {
...
-    tuple val(meta), path("*-busco.batch_summary.txt"), emit: batch_summary
-    tuple val(meta), path("short_summary.*.txt")      , emit: short_summaries_txt, optional: true
-    tuple val(meta), path("short_summary.*.json")     , emit: short_summaries_json, optional: true
-    tuple val(meta), path("*-busco")                  , emit: busco_dir
-    path "versions.yml"                               , emit: versions
+    tuple val(meta), path("*-busco.batch_summary.txt")      , emit: batch_summary
+    tuple val(meta), path("short_summary.*.txt")            , emit: short_summaries_txt, optional: true
+    tuple val(meta), path("short_summary.*.json")           , emit: short_summaries_json, optional: true
+    tuple val(meta), path("*-busco/*/run_*/full_table.tsv") , emit: full_table, optional: true
+    tuple val(meta), path("*-busco/*/run_*/busco_sequences"), emit: seq_dir
```
Comment (muffato) [inline modules/nf-core/busco/main.nf:21]: If run with the `--tar` option, I think this directory doesn't exist ? Or is Busco tar-ing the three sub-directories (fragmented_busco_sequences  multi_copy_busco_sequences  single_copy_busco_sequences) ?

```suggestion
    tuple val(meta), path("*-busco/*/run_*/busco_sequences"), emit: seq_dir, optional: true
```

**Review by muffato — COMMENTED** [review]
Can you add `missing_busco_list.tsv` ? That seems to be the only other main Busco file that could be used

File: modules/nf-core/busco/main.nf [hunk 4]
```diff
@@ -14,11 +14,16 @@ process BUSCO {
...
-    tuple val(meta), path("short_summary.*.json")     , emit: short_summaries_json, optional: true
-    tuple val(meta), path("*-busco")                  , emit: busco_dir
-    path "versions.yml"                               , emit: versions
+    tuple val(meta), path("*-busco.batch_summary.txt")                , emit: batch_summary
+    tuple val(meta), path("short_summary.*.txt")                      , emit: short_summaries_txt, optional: true
+    tuple val(meta), path("short_summary.*.json")                     , emit: short_summaries_json, optional: true
+    tuple val(meta), path("*-busco/*/run_*/full_table.tsv")           , emit: full_table, optional: true
+    tuple val(meta), path("*-busco/*/run_*/missing_busco_list.tsv")   , emit: missing_busco_list, optional: true
+    tuple val(meta), path("*-busco/*/run_*/single_copy_proteins.faa") , emit: single_copy_proteins, optional: true
+    tuple val(meta), path("*-busco/*/run_*/busco_sequences")          , emit: seq_dir
```
Comment (FriederikeHanssen) [inline modules/nf-core/busco/main.nf:23]: It's new but non-optional, is that righ?

Comment (muffato) [reply modules/nf-core/busco/main.nf:23]: Correct. It's present in all the runs I've seen so far, and all the combinations of options tested in nf-core.

**Review by FriederikeHanssen — APPROVED** [review]
Sorry  for the delay, I was on vacation 🌴

**Comment by muffato** [comment]
No worries, I took some holidays as well ;) Thank you !

**Pull request merged**